packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -239,18 +239,24 @@ export const buildAggregatedQuickFilterApplier = (
   );
 
   return (rowId, shouldApplyFilter) => {
+    const rowNode = apiRef.current.getRowNode(rowId);
+    if (rowNode?.isAutoGenerated) {
+      return true;
+    }
+
     const usedCellParams: { [field: string]: GridCellParams } = {};
+    const columnsFieldsToFilter: string[] = [];
 
     Object.keys(appliersPerColumnField).forEach((columnField) => {
       if (!shouldApplyFilter || shouldApplyFilter(columnField)) {
         usedCellParams[columnField] = apiRef.current.getCellParams(rowId, columnField);
+        columnsFieldsToFilter.push(columnField);
```

Comment:
The error comes from the use of `shouldApplyFilter`

the `usedCellParams[columnField]` was not set, if not necessary. But in the next block of code, I'm looping on `Object.keys(appliersPerColumnField)` to get the field

So I was calling `usedCellParams[columnField]` with some `columnField` not set